components/Banner/src/MDCBannerView.m

```diff
@@ -0,0 +1,474 @@
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *leadingButtonConstraintCenterY;
+@property(nonatomic, readwrite, strong)
+    NSLayoutConstraint *leadingButtonConstraintTrailingWithTrailingButton;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *trailingButtonConstraintBottom;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *trailingButtonConstraintTop;
+@property(nonatomic, readwrite, strong) NSLayoutConstraint *trailingButtonConstraintTrailing;
+
+@end
+
+@implementation MDCBannerView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (void)commonBannerViewInit {
+  self.backgroundColor = UIColor.whiteColor;
+
+  // Create textLabel
+  UILabel *textLabel = [[UILabel alloc] init];
+  textLabel.translatesAutoresizingMaskIntoConstraints = NO;
+  textLabel.font = [MDCTypography body2Font];
+  textLabel.textColor = UIColor.blackColor;
+  textLabel.alpha = [MDCTypography body2FontOpacity];
+  textLabel.numberOfLines = kTextNumberOfLineLimit;
+  textLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+  [self addSubview:textLabel];
+  _textLabel = textLabel;
+
+  // Create imageView
+  UIImageView *imageView = [[UIImageView alloc] init];
+  imageView.translatesAutoresizingMaskIntoConstraints = NO;
+  if (@available(iOS 9.0, *)) {
+    [imageView.widthAnchor constraintEqualToConstant:kImageViewSideLength].active = YES;
+    [imageView.heightAnchor constraintEqualToConstant:kImageViewSideLength].active = YES;
+  }
+  imageView.contentMode = UIViewContentModeCenter;
+  imageView.clipsToBounds = YES;
+  imageView.hidden = YES;
+  [self addSubview:imageView];
+  _imageView = imageView;
+
+  // Create a button container to organize buttons
+  UIView *buttonContainerView = [[UIView alloc] init];
+  buttonContainerView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self addSubview:buttonContainerView];
+  self.buttonContainerView = buttonContainerView;
+
+  // Create leadingButton and trailingButton
+  MDCButton *leadingButton = [[MDCButton alloc] init];
+  leadingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  leadingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:leadingButton];
+  _leadingButton = leadingButton;
+
+  MDCButton *trailingButton = [[MDCButton alloc] init];
+  trailingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  trailingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:trailingButton];
+  _trailingButton = trailingButton;
+
+  [self addObserversToSubviews];
+  [self setupConstraints];
+}
+
+#pragma mark - Constraints Helpers
+
+- (void)setupConstraints {
+  if (@available(iOS 9.0, *)) {
+    self.imageViewConstraintLeading =
```

Comment:
Please document each logical group of constraints in this method with a summary of each group of constraints and their purpose.

Reply:
Ah; I just noticed the comments near the top of this .m file which are helpful.

This method is still a bit logically complex, so it will benefit from some summarization in terms of what's going on.

Reply:
Consider breaking this method up into smaller named methods as a form of self-documentation.

Reply:
Sure! Addressed in bdbf5e0e1e0afc268adce436ed34138ebeda821e.